src/main/java/de/tum/in/www1/artemis/service/connectors/EmbeddingTrainingMaterialService.java

```diff
@@ -0,0 +1,85 @@
+package de.tum.in.www1.artemis.service.connectors;
+
+import static de.tum.in.www1.artemis.service.connectors.RemoteArtemisServiceConnector.authenticationHeaderForSecret;
+
+import java.io.IOException;
+
+import org.apache.commons.codec.binary.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Attachment;
+import de.tum.in.www1.artemis.exception.NetworkingError;
+import de.tum.in.www1.artemis.service.FileService;
+
+@Service
+@Profile("automaticText")
+public class EmbeddingTrainingMaterialService {
+
+    private final Logger log = LoggerFactory.getLogger(EmbeddingTrainingMaterialService.class);
+
+    // region Request/Response DTOs
+    private static class Request {
+
+        public long courseId;
+
+        public String fileName;
+
+        public String fileData;
+
+        public Request(long courseId, String fileName, String fileData) {
+            this.courseId = courseId;
+            this.fileName = fileName;
+            this.fileData = fileData;
+        }
+    }
+
+    private static class Response {
+
+        public String remotePath;
+    }
+    // endregion
+
+    @Value("${artemis.automatic-text.material-upload-url}")
+    private String API_ENDPOINT;
+
+    @Value("${artemis.automatic-text.secret}")
+    private String API_SECRET;
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968860)